src/parser/flow_lexer.ml

```diff
@@ -724,65 +767,86 @@ let token (env: Lex_env.t) lexbuf : result =
+  | octbigint ->
+    Token (env, T_BIGINT { kind = OCTAL; raw = lexeme lexbuf })
+  | (octnumber | octbigint), (letter | '8'..'9'), Star alphanumeric ->
     (* Numbers cannot be immediately followed by words *)
     recover env lexbuf ~f:(fun env lexbuf -> match%sedlex lexbuf with
     | octnumber -> Token (env, T_NUMBER { kind = OCTAL; raw = lexeme lexbuf })
+    | octbigint ->
+      Token (env, T_BIGINT { kind = OCTAL; raw = lexeme lexbuf })
     | _ -> failwith "unreachable"
     )
-
   | octnumber ->
     Token (env, T_NUMBER { kind = OCTAL; raw = lexeme lexbuf })
 
-  | legacyoctnumber, (letter | '8'..'9'), Star alphanumeric ->
+  | legacyoctbigint ->
+    Token (env, T_BIGINT { kind = LEGACY_OCTAL; raw = lexeme lexbuf })
+  | (legacyoctnumber | legacyoctbigint), (letter | '8'..'9'), Star alphanumeric ->
     (* Numbers cannot be immediately followed by words *)
     recover env lexbuf ~f:(fun env lexbuf -> match%sedlex lexbuf with
     | legacyoctnumber -> Token (env, T_NUMBER { kind = LEGACY_OCTAL; raw = lexeme lexbuf })
+    | legacyoctbigint -> Token (env, T_BIGINT { kind = LEGACY_OCTAL; raw = lexeme lexbuf })
     | _ -> failwith "unreachable"
     )
-
   | legacyoctnumber ->
     Token (env, T_NUMBER { kind = LEGACY_OCTAL; raw = lexeme lexbuf })
 
-  | hexnumber, non_hex_letter, Star alphanumeric ->
+  | hexbigint ->
+    Token (env, T_BIGINT { kind = NORMAL; raw = lexeme lexbuf })
+  | (hexnumber | hexbigint), non_hex_letter, Star alphanumeric ->
     (* Numbers cannot be immediately followed by words *)
     recover env lexbuf ~f:(fun env lexbuf -> match%sedlex lexbuf with
     | hexnumber -> Token (env, T_NUMBER { kind = NORMAL; raw = lexeme lexbuf })
+    | hexbigint -> Token (env, T_BIGINT { kind = NORMAL; raw = lexeme lexbuf })
     | _ -> failwith "unreachable"
     )
-
   | hexnumber ->
     Token (env, T_NUMBER { kind = NORMAL; raw = lexeme lexbuf })
 
-  | scinumber, word ->
+  | scibigint ->
+    let loc = loc_of_lexbuf env lexbuf in
+    let env = lex_error env loc Parse_error.InvalidBigInt in
+    Token (env, T_BIGINT { kind = NORMAL; raw = lexeme lexbuf })
+  | (scinumber | scibigint), word ->
     (* Numbers cannot be immediately followed by words *)
     recover env lexbuf ~f:(fun env lexbuf -> match%sedlex lexbuf with
     | scinumber -> Token (env, T_NUMBER { kind = NORMAL; raw = lexeme lexbuf })
+    | scibigint -> Token (env, T_BIGINT { kind = NORMAL; raw = lexeme lexbuf })
     | _ -> failwith "unreachable"
     )
-
   | scinumber ->
     Token (env, T_NUMBER { kind = NORMAL; raw = lexeme lexbuf })
 
-  | (wholenumber | floatnumber), word ->
+  | wholebigint ->
+    Token (env, T_BIGINT { kind = NORMAL; raw = lexeme lexbuf })
+  | floatbigint ->
+    let loc = loc_of_lexbuf env lexbuf in
+    let env = lex_error env loc Parse_error.InvalidBigInt in
+    Token (env, T_BIGINT { kind = NORMAL; raw = lexeme lexbuf })
+  | (wholenumber | floatnumber | wholebigint), word ->
```

Comment:
This is also ok to be one case. But you should add `floatbigint` and you should move this case before the others.